src/core/qgstemporalnavigationobject.h

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************
+                         qgstemporalnavigationobject.h
+                         ---------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSTEMPORALNAVIGATIONOBJECT_H
+#define QGSTEMPORALNAVIGATIONOBJECT_H
+
+#include "qgsrange.h"
+#include <QList>
+#include "qgstemporalcontroller.h"
+
+class QgsMapLayer;
+
+/**
+ * \ingroup core
+ * The QgsTemporalNavigationObject class
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsTemporalNavigationObject : public QgsTemporalController
+{
+
+  public:
+
+    /**
+      * Constructor for QgsTemporalNavigationObject
+      *
+      */
+    QgsTemporalNavigationObject();
+
+    //! Represents the current VCR navigation status
+    enum NavigationStatus
```

Comment:
I'd drop this. I think this class could be simplified by:

- add a private member `QgsDateTimeRange mTemporalExtents;`, with public getters and setters, for storing the earliest and latest possible datetime allowed during the navigation.
- rework the logic to use a "frame" based approach. Add a private member for "long long mCurrentFrameNumber", with public getters/setters (`setCurrentFrameNumber()`)
- add a member for the frame duration. This would be stored as a private QgsInterval mFrameDuration member, with public getters/setters
- add a public function `QgsDateTimeRange dateTimeRangeForFrameNumber( long long frame ) const;`, which would calculate the actual date time range for the given frame number, based on the start time from mTemporalExtents and mFrameDuration. 
- add a private member `double mFramesPerSecond`, with public getters/setters. This would control the overall frame rate of the animation
- Add a `QTimer* mNewFrameTimer` private member. The timer timeout would be set based on mFramesPerSecond (i.e. `1.0/mFramesPerSecond`), so it's firing every time a new frame is desired. 
- Add public slots for `play` and `pause`, which start or stop mNewFrameTimer respectively
- In setCurrentFrameNumber, you would use `dateTimeRangeForFrameNumber` to calculate the current date range, and then emit the updateTemporalRange() signal with the calculated range.
- remove the navigation status handling, and replace the enum based approach with a slot based approach. I.e. add slots for "forward" (skips forward one frame), "backward" (skips backward one frame), "rewindToStart" skips back to earliest possible date time, "skipToEnd" skips forward to latest possible date time. Each of these would just increment/decrement/set to 0/etc the current frame number and then call setCurrentFrameNumber with the new desired frame. (Which in turn causes the updateTemporalRange signal to be emitted)
- Connect the timeout signal from mNewFrameTimer to the forward slot

That should take care of everything :smile: